core/src/main/scala/cats/Cartesian.scala

```diff
@@ -16,4 +16,13 @@ import simulacrum.typeclass
   def product[A, B](fa: F[A], fb: F[B]): F[(A, B)]
 }
 
-object Cartesian extends CartesianArityFunctions
+object Cartesian extends CartesianArityFunctions with KernelCartesianInstances
+
+/**
+ * Cartesian instances for types that are housed in cats.kernel and therefore
+ * can't have instances for this type class in their companion objects.
+ */
+private[cats] sealed trait KernelCartesianInstances {
+  implicit val catsInvariantSemigroup: Cartesian[Semigroup] = InvariantMonoidal.catsInvariantMonoidalSemigroup
+  implicit val catsInvariantMonoid: Cartesian[Monoid] = InvariantMonoidal.catsInvariantMonoidalMonoid
```

Comment:
Actually I take that back, the `tut` compilation fails because it can't find `Cartesian[Semigroup]` with this change. I reverted it and made this commit for reference: https://github.com/OlivierBlanvillain/cats/commit/07d2c14471a0fa8f105961685f1612c28a3785af